pkg/controllers/capr/managesystemagent/managesystemagent.go

```diff
@@ -35,22 +36,43 @@ import (
 )
 
 const (
-	generationSecretName = "system-agent-upgrade-generation"
+	generationSecretName                  = "system-agent-upgrade-generation"
+	upgradeAPIVersion                     = "upgrade.cattle.io/v1"
+	upgradeDigestAnnotation               = "upgrade.cattle.io/digest"
+	systemAgentUpgraderServiceAccountName = "system-agent-upgrader"
 )
 
-var Kubernetes125 = semver.MustParse("v1.25.0")
+var (
+	Kubernetes125 = semver.MustParse("v1.25.0")
+
+	// GH5551FixedVersions is a slice of rke2 versions
+	// which have resolved GH-5551 for Windows nodes.
+	// ref:  https://github.com/rancher/rke2/issues/5551
+	// The SUC should not deploy plans to Windows nodes
+	// running a version less than the below for each minor.
+	// This check can be removed when 1.31.x is the lowest supported
+	// rke2 version.
+	GH5551FixedVersions = []*semver.Version{
```

Comment:
```suggestion
	GH5551FixedVersions = map[int]semver.Version{
```
This is totally a nit but would using a map here make sense? That way you could specify `currentVersionResolvesGH5551` as follows:
```
minor := curSemVer.Minor()
if minor >= 31 {
  return true
}
if minor <= 26 {
  return false
}
return curSemVer.GreaterThanEqual(GH5551FixedVersions(minor))
```

Reply:
It does kind of feel like we're duplicating information with the map, but I agree it makes the comparisons easier to read 👍